frontend/express/public/core/token-manager/javascripts/countly.views.js

```diff
@@ -0,0 +1,224 @@
+
+                endpoints.forEach(function(element) {
+                    if (element.endpointName !== "" && element.endpointName) {
+                        var obj = {params: {}} ;
+                        obj.endpoint = element.endpointName;
+                        element.parameters.forEach(function(item) {
+                            var key = item.queryParameters1;
+                            var value = item.queryParameters2;
+                            obj.params[key] = value;
+                        });
+                        newEndpoints.push(obj);
+                    }
+                });
+                endpoints = newEndpoints;
+                endpoints = JSON.stringify(endpoints);
+
+                if (doc.selectApps.length > 0) {
+                    selectApps = doc.selectApps.join(",");
+                }
+                if (doc.selectTime === "hours") {
+                    ttl = doc.timeInput * 3600;
+                }
+                else if (doc.selectTime === "days") {
+                    ttl = doc.timeInput * 3600 * 24;
+                }
+                else if (doc.selectTime === "months") {
+                    ttl = doc.timeInput * 3600 * 24 * 30;
+                }
+
+                countlyTokenManager.createTokenWithQuery(doc.description, endpoints, doc.checkboxMultipleTimes, selectApps, ttl, function() {
+                    self.$emit("create");
+                });
+            }
+        },
+        props: {
+            controls: {
+                type: Object
+            }
+        },
+    });
+
+    var TokenManager = countlyVue.views.create({
+        template: CV.T('/core/token-manager/templates/token-manager.html'),
+        components: {
+            "main-drawer": TokenDrawer
+        },
+        mixins: [countlyVue.mixins.hasDrawers("main")],
+        data: function() {
+            return {
+                tableData: []
+            };
+        },
+        mounted: function(row) {
+            var self = this;
+            $.when(countlyTokenManager.initialize()).then(function() {
+                self.prepareTableData(row);
+            });
+        },
+        methods: {
+            refresh: function(row) {
+                var self = this;
+                $.when(countlyTokenManager.initialize()).then(function() {
+                    self.prepareTableData(row);
+                });
+            },
+            prepareTableData: function(row) {
+                var tableData = countlyTokenManager.getData();
+                for (var j = 0; j < tableData.length; j++) {
+                    if (tableData[j]._id === countlyGlobal.auth_token) {
+                        tableData.splice(j, 1);
+                    }
+                }
+                for (var i = 0; i < tableData.length; i++) {
+                    row = tableData[i];
+                    if (row.ttl && ((row.ends * 1000) - Date.now()) < 0) {
+                        row.status = "Expired";
```

Comment:
We better use localizations for these :slightly_smiling_face: 

e.g. "token_manager.table.status-expired" for Expired

Reply:
Could you please also add an isExpired field to the row? :slightly_smiling_face: 

That would be useful for getColor method